src/main/java/io/github/thebusybiscuit/slimefun4/api/gps/TeleportationManager.java

```diff
@@ -58,10 +61,21 @@ public void openTeleporterGUI(Player p, UUID uuid, Block b, int complexity) {
                 }
 
                 int slot = teleporterInventory[index];
-
                 Location l = waypoint.getLocation();
-                menu.addItem(slot,
-                        new CustomItem(waypoint.getIcon(), waypoint.getName().replace("player:death ", ""), "", "&8\u21E8 &7" + SlimefunPlugin.getLocalization().getResourceString(p, "tooltips.world") + ": &f" + l.getWorld().getName(), "&8\u21E8 &7X: &f" + l.getX(), "&8\u21E8 &7Y: &f" + l.getY(), "&8\u21E8 &7Z: &f" + l.getZ(), "&8\u21E8 &7" + SlimefunPlugin.getLocalization().getMessage(p, "machines.TELEPORTER.gui.time") + ": &f" + DoubleHandler.fixDouble(0.5 * getTeleportationTime(complexity, source, l)) + "s", "", "&8\u21E8 &c" + SlimefunPlugin.getLocalization().getMessage(p, "machines.TELEPORTER.gui.tooltip")));
+                double time = DoubleHandler.fixDouble(0.5 * getTeleportationTime(complexity, source, l));
+
+                List<String> nameAndLore = new ArrayList<>();
+                nameAndLore.add(waypoint.getName().replace("player:death ", ""));
+                nameAndLore.add("");
+                nameAndLore.add("&8\u21E8 &7" + SlimefunPlugin.getLocalization().getResourceString(p, "tooltips.world") + ": &f" + l.getWorld().getName());
+                nameAndLore.add("&8\u21E8 &7X: &f" + l.getX());
+                nameAndLore.add("&8\u21E8 &7Y: &f" + l.getY());
+                nameAndLore.add("&8\u21E8 &7Z: &f" + l.getZ());
+                nameAndLore.add("&8\u21E8 &7" + SlimefunPlugin.getLocalization().getMessage(p, "machines.TELEPORTER.gui.time") + ": &f" + time + "s");
+                nameAndLore.add("");
+                nameAndLore.add("&8\u21E8 &c" + SlimefunPlugin.getLocalization().getMessage(p, "machines.TELEPORTER.gui.tooltip"));
```

Comment:
Probably best to use Arrays.asList

I can't see a reason why this should be done. Would be slower and means there's a mutable list (even if for a short time)